src/component/directive.ts

```diff
@@ -271,13 +271,39 @@ QWeb.addDirective({
     let tattClass = node.getAttribute("t-att-class");
     let styleAttr = node.getAttribute("style");
     let tattStyle = node.getAttribute("t-att-style");
+
+    const dataAttrs = [...node.attributes].filter(a => /^(t-attf?-)?data-.+$/.test(a.name)).sort();
+    if (dataAttrs.length) {
+      const dashToCamel = (str: string) => {
+        return str.toLowerCase()
+          .replace(/(\-[a-z])/g, val => val.toUpperCase().replace('-', ''));
+      };
+      ctx.addLine("let dataAttrMap = {};")
+      let attrName = "", attrValue = "";
```

Comment:
typically, we try not to do that, and rather define one variable per line (full definition: `let machin = "val";`)